src/chain/sync/grandpa_warp_sync.rs

```diff
@@ -155,6 +157,11 @@ impl<TSrc> StorageGet<TSrc> {
         &self.state.warp_sync_source
     }
 
+    /// Returns the header that we're warp syncing up to.
+    pub fn warp_sync_header(&self) -> Header {
+        self.state.header.clone()
+    }
```

Comment:
Let's please not just copy things all the time.

```suggestion
    pub fn warp_sync_header(&self) -> HeaderRef {
        (&self.state.header).into()
    }
```